yarp/yarp_compiler.c

```diff
@@ -1240,6 +1240,38 @@ yp_compile_node(rb_iseq_t *iseq, const yp_node_t *node, LINK_ANCHOR *const ret,
 
           return;
       }
+      case YP_NODE_SOURCE_ENCODING_NODE: {
+          const char *encoding = compile_context->parser->encoding.name;
+          if (!popped) {
+              rb_encoding *enc = rb_find_encoding(rb_str_new_cstr(encoding));
```

Comment:
This can return `NULL`. If it does, we should raise an `rb_bug`, because it shouldn't ever happen.